io/net/net_tools.py

```diff
@@ -0,0 +1,374 @@
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Basheer K<basheer@linux.vnet.ibm.com>
+#
+# Based on code by  Hong Bo Peng <penghb@cn.ibm.com>
+# copyright: 2003, 2015 IBM Corp
+
+import os
+import re
+import shutil
+from avocado import Test
+from avocado import main
+from avocado.core import data_dir
+from avocado.utils import process, distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+def install_dependencies():
+    """
+    To check and install  dependencies for the test
+    """
+    detected_distro = distro.detect()
+    sm = SoftwareManager()
+    if detected_distro.name == "SuSE":
+        net_tools = ("net-tools", "traceroute")
+    else:
+        net_tools = ("net-tools", "hostname", "traceroute")
+    for pkg in net_tools:
+        if not sm.check_installed(pkg) and not sm.install(pkg):
+            raise AssertionError("%s package is need to test" % pkg)
+
+
+class Hostname(Test):
+    """
+    hostname: set/get hostname by cmd. Check it with the expected result.
+    Test to verify the functionality of the hostname command
+    """
+
+    def setUp(self):
+        install_dependencies()
+        self.restore_hostname = False
+        # Get Hostname
+        output = process.run("hostname", verbose=False,
+                             env={"LANG": "C"})
+        if not output.stdout:
+            # set hostname if not set
+            if not process.system("hostname localhost.localdomain", sudo=True):
+                self.error("Failed to set hostname\n")
+
+        self.hostname = process.run("hostname", verbose=False,
+                                    env={"LANG": "C"}).stdout
+        # Backup info required to restore hostname
+        self.workdir = data_dir.get_tmp_dir()
+        self.backup_dir = os.path.join(self.workdir, 'net-tools')
+        if not os.path.isdir(self.backup_dir):
+            os.makedirs(self.backup_dir)
+
+        self.hostname_file = os.path.join(self.backup_dir, "HOSTNAME")
+        self.myhostname_file = os.path.join(self.backup_dir, "MYHOSTNAME")
+        fobj = open(self.hostname_file, 'w')
+        fobj.write(self.hostname)
+        fobj.close()
+        fobj = open(self.myhostname_file, 'w')
+        fobj.write("myhost.my-domain")
+        fobj.close()
+
+    def test_hostname(self):
+        """
+        i.Verifies hostname command options and
+        ii.Test to change the hostname
+        """
+        output = process.run("hostname", verbose=False,
+                             env={"LANG": "C"})
+        if not output.stdout:
+            self.error("unexpected response from hostname command")
+
+        # Verifying different options provided by hostname
+        options_to_verify = self.params.get('hostname_opt', default="f")
+        for option in options_to_verify:
+            output = process.run("hostname -%s" % option, verbose=False)
+            if not output.stdout:
+                self.error("unexpected response from hostname command")
```

Comment:
Actually there are lot of `process.run`s which are basically equal to test failure. So instead of treating each one I'd recommend using fail_on decorator:

```python
import avocado
...
@avocado.fail_on(process.CmdError)
def test_hostname(self):
    ...
    if not process.system_output("hostname -%s" % option):
        self.fail("No output for %s option" % option)
```

And similarly for other `process` executions. The `fail_on` decorator turns the `process.CmdError` exceptions into test failures.